scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,284 @@
+        "armedForceStatus": "Civilian"
+      },
+      "<column-name-2>": {}, .....
+    }
+  """
+    col_map_obj = GenerateColMapBase(spec_dict=spec_dict,
+                                     column_list=column_list)
+    return col_map_obj._generate_stat_vars_from_spec()
+
+
+class GenerateColMapBase:
+    """module to generate a column map given a list of columns of the dataset and a JSON Spec
+
+  Attributes:
+    specDict: A dictionary containing specifications for the different properties of the statistical variable.
+    columnList: A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+    delimiter: The delimiting string that is used for tokenising the column name
+  """
+
+    def __init__(self, spec_dict={}, column_list=[], delimiter='!!'):
+        """module init"""
+        self.features = spec_dict
+        self.column_list = column_list
+        self.column_map = {}
+        self.delimiter = delimiter
+
+        # fill missing keys in JSON spec with empty values
+        for key in self.features:
+            try:
+                self.features[key]
+            except KeyError:
+                self.features[key] = {}
+
+    def _generate_stat_vars_from_spec(self):
+        """generates stat_var nodes for each column in column list and returns a dictionary of columns with their respective stat_var nodes
+        Example output: {
+      "Total Civilian population": {
+        "populationType": "Person",
+        "statType": "measuredValue",
+        "measuredProperty": "Count Person"
+        "armedForceStatus": "Civilian"
+      },
+      "<column-name-2>": {}, .....
+    }"""
+        # for each column generate the definition of their respective statistical variable node
+        for col in self.column_list:
+            if 'ignoreColumns' in self.features and col not in self.features[
+                    'ignoreColumns']:
+                try:
+                    self.column_map[col] = self._column_to_statvar(col)
+                except Exception as e:
+                    exec_type, exec_obj, exec_tb = sys.exc_info()
+                    logger.error(
+                        f"Exeception: {exec_type} occured in generate_col_map.py at line {exec_tb.tb_lineno}"
+                    )
+                    continue
+
+        #TODO: Enable validating the generate columnMap. Dependent implementation _isvalid_column_map
+        return self.column_map
+
+    def _column_to_statvar(self, column):
+        """generates a dictionary statistical variable with all properties specified in the JSON spec for a single column"""
+        measurement_assigned = False
+        stat_var = {}
+        # part_list contains a list of tokens of the column name after splitting the
+        # column name based on the '!!' delimiter
+        part_list = column.split(self.delimiter)
+
+        # part_list[-1], the last element of part_list describes the property value.
```

Comment:
I'm still not sure about this assumption.  I responded to your table-shell comment in https://github.com/datacommonsorg/data/pull/488#discussion_r700728539, which you resolved without response, so reopening...